### PR TITLE
Update FreeBSD from 12.2 to 13.2

### PR DESCRIPTION
**Pull request by ararslan**
The FreeBSD toolchain will need to be updated before this is ready, opening now as draft so I don't forget.

### CODE REVIEW

File: 0_RootFS/gcc_sources.jl [hunk 1]
```diff
@@ -250,8 +250,8 @@ function gcc_sources(gcc_version::VersionNumber, compiler_target::Platform; kwar
         end
     elseif Sys.isfreebsd(compiler_target)
         libc_sources = [
-            ArchiveSource("http://ftp-archive.freebsd.org/pub/FreeBSD-Archive/old-releases/amd64/12.2-RELEASE/base.txz",
-                          "8bd49ce35c340a04029266fbbe82b1fdfeb914263e39579eecafb2e67d00693a"),
+            ArchiveSource("https://download.freebsd.org/ftp/releases/amd64/amd64/13.2-RELEASE/base.txz",
```
Comment (ararslan) [inline 0_RootFS/gcc_sources.jl:253]:
```suggestion
            ArchiveSource("http://ftp-archive.freebsd.org/pub/FreeBSD-Archive/old-releases/amd64/13.2-RELEASE/base.txz",
```

File: 0_RootFS/gcc_sources.jl [hunk 2]
```diff
@@ -250,8 +250,8 @@ function gcc_sources(gcc_version::VersionNumber, compiler_target::Platform; kwar
...
+                          "3a9250f7afd730bbe274691859756948b3c57a99bcda30d65d46ae30025906f0"),
```
[the remark this answers is not in the document]
Comment (ararslan) [reply 0_RootFS/gcc_sources.jl]: Right, the non-archive links get removed once a release is EOL. I assumed that current releases were not in `FreeBSD-Archive/old-releases` but I was incorrect. I've updated to use that.

**Comment by ararslan** [comment]
Is the right way to update the BinaryBuilderBase Artifacts.toml still what's described in the comment at the top of `0_RootFS/PlatformSupport/build_tarballs.jl`?

**Comment by giordano** [comment]
Yes, that has to be done separately.  It'd also be great to take notes of all the steps that you're doing to update this, because this is what we need to do to add a new platform to Julia in general (CC: @gbaraldi)

**Comment by ararslan** [comment]
Breadcrumbs for myself from the last rodeo:
- https://github.com/JuliaPackaging/Yggdrasil/pull/3447
- https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/166

**Comment by ararslan** [comment]
I got PlatformSupport deployed successfully but ran into an error when doing GCCBootstrap@4:
```
[ Info: GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.squashfs hash: 2d57fa515d63d3ab4f48a086f31fe89a092e2a0d
WARNING: found release (GCCBootstrap-v4.8.5+2). Use existing one.
--> Uploading: GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz
Failed to upload one of assets: one of the goroutines failed: failed to upload asset: /tmp/jl_8KhAvk/GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpa
cked.tar.gz: failed to upload release asset: /tmp/jl_8KhAvk/GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz: POST https://uploads.github.
com/repos/JuliaPackaging/Yggdrasil/releases/27708172/assets?name=GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz: 422 Validation Failed [
{Resource:ReleaseAsset Field:name Code:already_exists Message:}]
WARNING: found release (GCCBootstrap-v4.8.5+2). Use existing one.
--> Uploading: GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz
Failed to upload one of assets: one of the goroutines failed: failed to upload asset: /tmp/jl_8KhAvk/GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpa
cked.tar.gz: failed to upload release asset: /tmp/jl_8KhAvk/GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz: Post "https://uploads.github
.com/repos/JuliaPackaging/Yggdrasil/releases/27708172/assets?name=GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz": http2: Transport: can
not retry err [stream error: stream ID 5; REFUSED_STREAM] after Request.Body was written; define Request.GetBody to avoid this error
WARNING: found release (GCCBootstrap-v4.8.5+2). Use existing one.
--> Uploading: GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz
Failed to upload one of assets: one of the goroutines failed: failed to upload asset: /tmp/jl_8KhAvk/GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpa
cked.tar.gz: failed to upload release asset: /tmp/jl_8KhAvk/GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz: POST https://uploads.github.
com/repos/JuliaPackaging/Yggdrasil/releases/27708172/assets?name=GCCBootstrap-x86_64-linux-gnu.v4.8.5.x86_64-linux-musl.unpacked.tar.gz: 422 Validation Failed [
{Resource:ReleaseAsset Field:name Code:already_exists Message:}]
ERROR: LoadError: Unable to upload /tmp/jl_8KhAvk to GitHub repo JuliaPackaging/Yggdrasil on tag GCCBootstrap-v4.8.5+2
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] upload_to_github_releases(repo::String, tag::String, path::String; gh_auth::GitHub.OAuth2, attempts::Int64, verbose::Bool)
    @ BinaryBuilder ~/.julia/dev/BinaryBuilder/src/AutoBuild.jl:497
  [3] upload_to_github_releases(repo::String, tag::String, path::String)
    @ BinaryBuilder ~/.julia/dev/BinaryBuilder/src/AutoBuild.jl:483
  [4] (::var"#9#10"{String, String, Base.SHA1, String})(dir::String)
    @ Main ~/test/Yggdrasil/0_RootFS/common.jl:54
  [5] mktempdir(fn::var"#9#10"{String, String, Base.SHA1, String}, parent::String; prefix::String)
    @ Base.Filesystem ./file.jl:760
  [6] mktempdir (repeats 2 times)
    @ ./file.jl:756 [inlined]
  [7] publish_artifact
    @ ~/test/Yggdrasil/0_RootFS/common.jl:49 [inlined]
  [8] upload_compiler_shard(repo::String, name::String, version::VersionNumber, hash::Base.SHA1, archive_type::Symbol; platform::Platform, target::Platform)
    @ Main ~/test/Yggdrasil/0_RootFS/common.jl:92
  [9] upload_compiler_shard
    @ ~/test/Yggdrasil/0_RootFS/common.jl:88 [inlined]
 [10] upload_and_insert_shards(repo::String, name::String, version::VersionNumber, unpacked_hash::Base.SHA1, squashfs_hash::Base.SHA1, platform::Platform; targe
t::Platform)
    @ Main ~/test/Yggdrasil/0_RootFS/common.jl:123
 [11] upload_and_insert_shards
    @ ~/test/Yggdrasil/0_RootFS/common.jl:118 [inlined]
 [12] upload_and_insert_shards(repo::String, name::String, version::VersionNumber, build_info::Dict{Platform, Tuple{String, String, Base.SHA1, Dict{Product, Any
}}}; target::Platform)
    @ Main ~/test/Yggdrasil/0_RootFS/common.jl:137
 [13] build_and_upload_gcc(version::VersionNumber, ARGS::Vector{String})
    @ Main ~/test/Yggdrasil/0_RootFS/gcc_common.jl:604
 [14] build_and_upload_gcc(version::VersionNumber)
    @ Main ~/test/Yggdrasil/0_RootFS/gcc_common.jl:585
 [15] top-level scope
    @ ~/test/Yggdrasil/0_RootFS/GCCBootstrap@4/build_tarballs.jl:2
in expression starting at /home/ararslan/test/Yggdrasil/0_RootFS/GCCBootstrap@4/build_tarballs.jl:2
```

**Comment by ararslan** [comment]
IIUC, I think Polymake and Zig need an updated Perl in order to build, which means they can't be updated as part of this PR. I'm not sure why Perl is failing; does that also need to be separate?

**Comment by ararslan** [comment]
Aha, I missed the part in the log that mentioned PlatformSupport, thanks! I assume then https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/312 will need to be merged first in order for it to recognize the more recent PlatformSupport version?

**Comment by ararslan** [comment]
```
Downloading artifact: PlatformSupport-x86_64-unknown-freebsd12.2.v2021.8.10.x86_64-linux-musl.unpacked
```
It isn't picking up the 13.2 one. 😕 I guess I needed to remove 12.2 from the BinaryBuilderBase Artifacts.toml after all? @giordano, do you know?

**Comment by giordano** [comment]
You need to update the manifest at https://github.com/JuliaPackaging/Yggdrasil/actions/workflows/update_manifest.yml, merge the PR that's opened by the bot, and then rebase this one on master

**Comment by ararslan** [comment]
Looks like things are good for the most part except Polymake does indeed seem like it will need the Perl JLL updated first.

**Comment by ararslan** [comment]
Would the JLLs need to have their compatibility bounds adjusted if the Perl JLL changes the name of one of the files that's being referenced by Polymake and Zig? If so, that's quite annoying, so to avoid having to do that again we could make Perl name the system-specific configuration file based on `echo "${target}" | sed -E 's/[0-9.]+$//g'` rather than just `${target}` so that `x86_64-unknown-freebsd12.2` and `x86_64-unknown-freebsd13.2` would both just map to `x86_64-unknown-freebsd`.

**Comment by ararslan** [comment]
@benlorenz, Perl_jll v5.34.1 is now available! Thanks so much for moving that along.

**Comment by ararslan** [comment]
I rebased on master to pick up the Perl stuff and I dropped the polymake stuff from this PR since it will be handled separately. I left in the Zig change, which I can separate if that'd be preferable, but otherwise I think this is now good to go!

**Comment by ararslan** [comment]
Bump

**Pull request merged**